HD/Language_HDC/Language_HDC.md

```diff
@@ -0,0 +1,117 @@
+
+* **Related works**
+    * Hyperdimensional computing: Brain inspired computing model using vectors of very hihg dimensionality. It has features such as robust in noise and extremly low latency computation.
+    * Vector Symbolic Architecture: A well defined set of operation on large vector, in HDC bundling, binding and perutation are used to combine, bind information.
+    * Memory Centric Design: Hardware design technique where computation components are build in and around memory to bypass the expensive and high latency data transfer limitation of traditional computer architechture.
+
+* **Challenges**
+    * A hardware design cover all componenets required by HDC classification task.
+    * Properly define operations on hypervector (Bundling, Binding, Permutation)
+    * Efficient implementation on those operations with fast computation and memory access.
+    * Gain efficiency with minimum loss in accuracy.
+
+### Method
+
+* **HD Operation and Encoding Process**
+  * Bundling: Compoenet-wise addition.
+  * Binding: Compoenet-wise XOR.
+  * Permutation: Compoenet-wise rotate.
+  * For efficiency and computation simplicity, all hypervector are in binary form. \
+    ![Encoding](./Encoding.PNG)
+  * To preserve sequential info, binding and permutation are applied.
+  * Classification are done with cosine similarity check. \
+    ![Similarity](./Similarity.PNG)
+
+![Overall_Design](./System_Design.PNG)
+* **Hardware Design**
+  * HDC classification mimimally requires encoder and simularity checker to carry out classification task.
+    * In Language Classification:
+      * Trigram are input, querry corresponding letter hypervector and apply binding and permutation.
+      * All trigram in the entire text are bundled in a accmulator.
+      * After accmulated all trigram, binarise the resulting hypervector with some threshold k.
+
+
+### Evaluations
+
+* **Evaluation Setup**
+  * Test on Eurpoean Language dataset
+    * Texts Sample of 21 different language
+  * Baseline: KNN clustering with N-gram Histogram.
+
+
+
+* **Evaluation Result**
+  ![Result](./Result.PNG)
+  * Result shows proposed hardware design can greatly acclerate HDC computation with small accuracy trade-off.
+  * Memory usage increases exponetially in baseline method, whereas HDC requires constant memory.
+  * Robust against faulty memory cell due to high dimensionality and holographic representation of HDC.
+
+
+### Pros and Cons
+
+* Pros:
+  * Generic design, can generalize to other applications.
+  * Mnemonic centric, bypass BUS data transfer limitation.
+  * Take faulty hardware into consideration.
+
+
+* Cons:
+  * Simularity check seems computationally complex, why not hamming distance.
+  * Arbitrary Spares/Dense hypervector decison with no explaination.
```

Comment:
typo: spares -> sparse, decison -> decision, explaination -> explanation

Reply:
Sorry for all the typo....